File: scratch/rename_to_docszar.js
const fs = require('fs');
const path = require('path');

const replacements = [
  ["Zargigs Technologies", "DocsZar Technologies"],
  ["zargigstechnologies@gmail.com", "contactdocszar@gmail.com"],
  ["contactzargigs@gmail.com", "contactdocszar@gmail.com"],
  ["contactgigsflix@gmail.com", "contactdocszar@gmail.com"],
  ["gigsflixtechnologies@gmail.com", "contactdocszar@gmail.com"],
  ["ZARGIGS ENTERPRISE", "DOCSZAR ENTERPRISE"],
  ["ZARGIGS", "DOCSZAR"],
  ["Zargigs", "DocsZar"],
  ["zargigs", "docszar"],
  ["GigsFlix", "DocsZar"],
  ["Gigsflix", "DocsZar"],
  ["gigsflix.com", "docszar.com"],
  ["zargigs.com", "docszar.com"]
];

function processDir(dir) {
  const entries = fs.readdirSync(dir, { withFileTypes: true });
  for (const entry of entries) {
    const fullPath = path.join(dir, entry.name);
    if (entry.isDirectory()) {
      if (entry.name !== 'node_modules' && entry.name !== 'dist' && entry.name !== '.git') {
        processDir(fullPath);
      }
    } else if (entry.isFile()) {
      const ext = path.extname(entry.name).toLowerCase();
      if (['.jsx', '.js', '.html', '.json', '.sql', '.ts', '.md', '.css'].includes(ext)) {
        let content = fs.readFileSync(fullPath, 'utf8');
        let orig = content;
        for (const [oldVal, newVal] of replacements) {
          content = content.split(oldVal).join(newVal);
        }
        if (content !== orig) {
          fs.writeFileSync(fullPath, content, 'utf8');
          console.log(`Updated: ${fullPath}`);
        }
      }
    }
  }
}

['frontend', 'supabase'].forEach(dir => {
  if (fs.existsSync(dir)) processDir(dir);
});

console.log('Renaming to DocsZar completed!');

File: scratch/rename_to_docszar.py
import os
import re

replacements = [
    # Exact cases
    ("Zargigs Technologies", "DocsZar Technologies"),
    ("zargigstechnologies@gmail.com", "contactdocszar@gmail.com"),
    ("contactzargigs@gmail.com", "contactdocszar@gmail.com"),
    ("contactgigsflix@gmail.com", "contactdocszar@gmail.com"),
    ("gigsflixtechnologies@gmail.com", "contactdocszar@gmail.com"),
    ("ZARGIGS ENTERPRISE", "DOCSZAR ENTERPRISE"),
    ("ZARGIGS", "DOCSZAR"),
    ("Zargigs", "DocsZar"),
    ("zargigs", "docszar"),
    ("GigsFlix", "DocsZar"),
    ("Gigsflix", "DocsZar"),
    ("gigsflix.com", "docszar.com"),
    ("zargigs.com", "docszar.com"),
]

def update_file(filepath):
    try:
        with open(filepath, 'r', encoding='utf-8') as f:
            content = f.read()
        
        orig_content = content
        for old, new in replacements:
            content = content.replace(old, new)
            
        if orig_content != content:
            with open(filepath, 'w', encoding='utf-8') as f:
                f.write(content)
            print(f"Updated: {filepath}")
    except Exception as e:
        print(f"Error updating {filepath}: {e}")

targets = ["frontend", "supabase"]
for target in targets:
    for root, dirs, files in os.walk(target):
        if "node_modules" in dirs:
            dirs.remove("node_modules")
        if "dist" in dirs:
            dirs.remove("dist")
        if ".git" in dirs:
            dirs.remove(".git")
        for file in files:
            if file.endswith((".jsx", ".js", ".html", ".json", ".sql", ".ts", ".md", ".css")):
                update_file(os.path.join(root, file))

print("Completed renaming to DocsZar.")
